apps/studio/components/interfaces/Organization/BillingSettings/BillingBreakdown/UpcomingInvoice.tsx

```diff
@@ -312,4 +309,16 @@ const InvoiceTooltip = ({ text, linkRef }: { text: string; linkRef?: string }) =
   )
 }
 
+const formatToCurrency = (amount: number | undefined | null): string | null => {
+  if (amount === undefined || amount === null) {
+    return null
+  } else {
+    return Intl.NumberFormat('en-US', {
```

Comment:
Initializing Intl.NumberFormat is quite expensive, given we re-use it quite a lot + all the re-rendering, let's move

```ts
Intl.NumberFormat('en-US', {
      style: 'currency',
      currency: 'USD',
      minimumFractionDigits: 0,
    })
```

to a constant outside of this function

Reply:
Could also extract this NumberFormat + the formatToCurrency function to some utils, as we can re-use that nicely on other pages

Reply:
Done